AudioPlayer/AudioPlayer/AudioPlayer.swift

```diff
@@ -601,11 +601,69 @@ public class AudioPlayer: NSObject {
     - parameter time: The time to seek to.
     */
     public func seekToTime(time: NSTimeInterval) {
-        player?.seekToTime(CMTimeMake(Int64(time), 1))
+        let time = CMTime(seconds: time, preferredTimescale: 1000000000)
+        let seekableRange = player?.currentItem?.seekableTimeRanges.last?.CMTimeRangeValue
+        let seekableStart = seekableRange!.start
+        let seekableEnd = seekableRange!.end
+        
+        // check if time is in seekable range
+        if time >= seekableStart && time <= seekableEnd {
+            // time is in seekable range
+            player?.seekToTime(time)
+        } else if time < seekableStart {
+            // time is before seekable start, so just move to the most early position as possible
+            seekToStart()
+        } else if time > seekableEnd {
+            // time is larger than possibly, so just move forward as far as possible
+            seekToLive()
+        }
+
+        updateNowPlayingInfoCenter()
+    }
+    
+    /**
+     Seeks forward as far as possible
+     
+     */
+    public func seekToLive() {
```

Comment:
I thought about that. However currently `seekToStart()` might not do what somebody expects from the method name. It does not seek to the start. It seeks as far back as possible including a padding time.

Maybe you know a better method name to represent that, so it is clearer for the API users what it does actually.